test/it/uk/gov/hmrc/thirdpartyapplication/repository/ApplicationRepositorySpec.scala

```diff
@@ -102,13 +103,46 @@ class ApplicationRepositorySpec extends UnitSpec with MongoSpecSupport
 
       val application =
         anApplicationData(applicationId, "aaa", "111", productionState("requestorEmail@example.com"))
-          .copy(lastAccess = DateTime.now.minusDays(20)) // scalastyle:ignore magic.number
+          .copy(lastAccess = Some(DateTime.now.minusDays(20))) // scalastyle:ignore magic.number
 
       await(applicationRepository.save(application))
 
       val retrieved = await(applicationRepository.recordApplicationUsage(applicationId))
 
-      retrieved.lastAccess.isAfter(testStartTime) shouldBe true
+      retrieved.lastAccess.get.isAfter(testStartTime) shouldBe true
+    }
+  }
+
+  "setMissingLastAccessedDates" should {
+    def applicationWithoutLastAccessDate(applicationId: UUID) = {
+      await(applicationRepository.save(anApplicationData(applicationId, UUID.randomUUID().toString, UUID.randomUUID().toString)))
+
+      def applicationById: JsObject = Json.obj("id" -> applicationId.toString)
+      def removeLastAccessField: JsObject = Json.obj("$unset" -> Json.obj("lastAccess" -> ""))
+
+      applicationRepository.findAndUpdate(applicationById, removeLastAccessField)
+    }
+
+    "set lastAccess property on applications that do not have it" in {
+      val application1Id = UUID.randomUUID()
+      val application2Id = UUID.randomUUID()
+      val dateToSet: DateTime = DateTime.now
+
+      await(applicationWithoutLastAccessDate(application1Id))
+      await(applicationWithoutLastAccessDate(application2Id))
+      await(applicationRepository.save(anApplicationData(UUID.randomUUID()))) // Application with lastAccess date
```

Comment:
Is it worth having a third application that does have a last access date and can be checked for being unchanged?

Reply:
Do have one - the third item there (line 133) :-)

Reply:
Oh yeah! But we don't check for it being unchanged, we only check the count, which is not quite the same.